pkg/unidling/controller/controller.go

```diff
@@ -61,6 +63,68 @@ func (c *lastFiredCache) AddIfNewer(info types.NamespacedName, newLastFired time
 	return false
 }
 
+// ScaleUpdater implements unidlingutil.ScaleUpdater
+type ScaleUpdater struct {
+	Namespace string
+	DCGetter  deployclient.DeploymentConfigsGetter
+	RCGetter  kclient.ReplicationControllersGetter
+}
+
+func (s ScaleUpdater) Update(annotator *unidlingutil.ScaleAnnotater, obj runtime.Object, scale *kextapi.Scale) error {
+	switch typedObj := obj.(type) {
+	case *deployapi.DeploymentConfig:
+		if typedObj.Annotations == nil {
+			typedObj.Annotations = make(map[string]string)
+		}
+
+		originalObj, err := json.Marshal(typedObj)
+		if err != nil {
+			return err
+		}
+
+		annotator.ChangeAnnotations(typedObj.Spec.Replicas, typedObj.Annotations)
+		typedObj.Spec.Replicas = scale.Spec.Replicas
+
+		newObj, err := json.Marshal(typedObj)
```

Comment:
@soltysh
> That wasn't still addressed, we should not be using raw json package for any kind of encoding. Sooner or later this will bite us hard. I'd rather fix it before that happens. Here and the next call in the following case, as well.

Thanks for the feedback, done!